subway_prediction_ml/model_evaluator.py

```diff
@@ -0,0 +1,60 @@
+import numpy as np
+from sklearn.metrics import mean_squared_error
+
+
+class ModelEvaluator:
+    def __init__(self, model, features, labels):
+        self.model = model
+        self.features = features
+        self.labels = labels
+        self.predictions = self.model.predict(self.features)
+        self._binned_accuracy = None
+        self._rmse = None
+
+    def binned_accuracy(self):
+        self._binned_accuracy = self._binned_accuracy or \
+            self._calculate_binned_accuracy()
+        return self._binned_accuracy
+
+    def rmse(self):
+        self._rmse = self._rmse or self._calculate_rmse()
+        return self._rmse
+
+    def _calculate_binned_accuracy(self):
+        min_accuracy_bounds = map(self._min_accuracy_bounds, self.labels)
+        max_accuracy_bounds = map(self._max_accuracy_bounds, self.labels)
+        errors = self.predictions - self.labels
+        self._in_bounds = [
+            error >= min and error <= max
+            for (error, min, max) in zip(
+                errors,
+                min_accuracy_bounds,
+                max_accuracy_bounds
+            )
+        ]
+        n_within_bounds = len(list(filter(lambda x: x, self._in_bounds))) * 1.0
```

Comment:
I assume the `* 1.0` is to cast to a float? Should that maybe done on the next line where the division actually happens?